lib/adapters/http/read.js

```diff
@@ -88,13 +90,18 @@ var Read = Juttle.proc.source.extend({
                 res.on('end', function() {
                     self.triggerFailure('StatusCodeError: ' + res.statusCode +
                                         ' - ' + body.join(''));
+                    self.emit_eof();
                 });
             } else {
                 var format;
-                try {
-                    format  = contentType.parse(res).type.split('/')[1];
-                } catch(err) {
-                    // type is undefined
+                if (self.format) {
+                    format = self.format;
+                } else {
+                    try {
+                        format  = contentType.parse(res).type.split('/')[1];
+                    } catch(err) {
+                        // type is undefined
+                    }
                 }
 
                 try {
```

Comment:
You can use `Promise.try` here so you don't have to duplicate the sync/async error handling.